ProjectPi/src/helper/Converter.java

```diff
@@ -0,0 +1,180 @@
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class Converter {
+
+	/**
+	 * Reads the entire file at once
+	 */
+	public static byte[] fileToBytes(File file) {
+		return fileToBytes(file, (int) file.length(), false, false);
+	}
+
+	/**
+	 * Convert file to array of bytes in which each block of "header" + mtu bytes
+	 * represents a packet
+	 */
+	public static byte[] fileToPacketByteArray(File file) {
+		return fileToBytes(file, Transfer.getMTU(), true, true);
+	}
+
+	/**
+	 * Convert file to array of bytes
+	 */
+	public static byte[] fileToBytes(File file, int readSize, boolean withSeqNr, boolean withChecksum) {
+		FileInputStream fis = null;
+		try {
+			fis = new FileInputStream(file);
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		}
+
+		int len = (int) file.length();
+		if (withSeqNr || withChecksum) {
+			len += Transfer.getHeadSize() * (len / readSize + 1);
+		}
+		byte[] byteArray = new byte[len];
+
+		int off = 0;
+		int seqNr = 0;
+		int checksum = 0;
+
+		int maxSeqNr = (int) (Math.pow(2, 7));
+
+		while (off < len) {
+
+			if (withSeqNr || withChecksum) {
+				byteArray[off] = (byte) seqNr;
+				byteArray[off + 1] = 0;
+				off += Transfer.getHeadSize();
+				seqNr++;
+				if (seqNr == maxSeqNr) {
+					seqNr = 0;
+				}
+			}
+
+			try {
+				fis.read(byteArray, off, Math.min(readSize, (len - off)));
+				if (withChecksum) {
+					// Calculate the checksum for the part that has just been read
+					// and update the last two checksum bytes
+					byte[] dataArray = new byte[Math.min(readSize, (len - off))];
+					System.arraycopy(byteArray, off, dataArray, 0, Math.min(readSize, (len - off)));
+					checksum = calcChecksum(dataArray);
+					byteArray[off - 2] = (byte) ((checksum >>> 8) & 0xFF);
+					byteArray[off - 1] = (byte) (checksum & 0xFF);
+				} else {
+					byteArray[off - 2] = (byte) 0;
+					byteArray[off - 1] = (byte) 0;
+				}
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+
+			off += readSize;
+
+		}
+		try {
+			fis.close();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return byteArray;
+	}
+
+	/**
+	 * Calculates the checksum for a packet with data
+	 */
+	public static int calcChecksum(byte[] dataArray) {
+		int checksum = 0;
+		byte[] tempArray = new byte[2];
+		for (int i = 0; i < dataArray.length - 1; i += 2) {
+			tempArray[0] = dataArray[i];
+			tempArray[1] = dataArray[i + 1];
+			checksum += ((tempArray[1] & 0xFF) << 8) | ((tempArray[0] & 0xFF) << 0);
+		}
+		checksum = checksum & 32767;
+		return checksum;
+	}
+
+	/**
+	 * Convert array of bytes to file and return that file
+	 */
+	public static File bytesToFile(byte[] byteArray, String pathName) {
+		FileOutputStream fos = null;
+		File file = new File(pathName);
+		try {
+			fos = new FileOutputStream(file);
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
```

Comment:
This is really not useful error handling (except while debugging :-) ). Also because after the stack trace is printed, I guess the program resumes its execution? (or does the `printStackTrace` method also abort the program?). If it is resumed, the next try/catch raise/catch.